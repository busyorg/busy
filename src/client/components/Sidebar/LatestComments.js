import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative } from 'react-intl';
import { Link } from 'react-router-dom';
import BodyShort from '../Story/BodyShort';
import './LatestComments.less';

const RenderComment = ({ text, author, created }) =>
  (<div className="Comment">
    <div className="Comment__text">
      <div className="Comment__content">
        <BodyShort body={text} length={80} />
      </div>
      <div className="Comment__footer">
        <Link to={`/@${author}`} className="LatestComment__author">
          {author}
        </Link>
        <span className="LatestComment__bullet" />
        <span className="LatestComment__date">
          <FormattedRelative value={`${created}Z`} />
        </span>
      </div>
    </div>
  </div>);

RenderComment.propTypes = {
  text: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  created: PropTypes.string.isRequired,
};

const LatestComments = ({ comments }) =>
  (<div className="LatestComments">
    <div className="LatestComments__container">
      <h4 className="LatestComments__title">
        <i className="iconfont icon-time LatestComments__icon" />
        {' '}
        <FormattedMessage id="latest_comments" defaultMessage="Latest Comments" />
      </h4>
      <div className="LatestComments__divider" />
      {comments && comments.map(comment => <RenderComment key={comment.id} {...comment} />)}
      <div className="LatestComments__divider" />
      <h4 className="LatestComments__more">
        <Link to={'/latest-comments'}>
          <FormattedMessage id="see_all_comments" defaultMessage="See All Comments" />
        </Link>
      </h4>
    </div>
  </div>);

LatestComments.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape()),
};

LatestComments.defaultProps = {
  comments: [],
};

export default LatestComments;
