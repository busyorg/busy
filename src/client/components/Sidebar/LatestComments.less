@import (reference) "../../styles/custom.less";

.LatestComments {
  background: @white;
  border: 1px solid #e9e7e7;
  border-radius: @border-radius-base;
  &__container {
    padding: 8px 16px 16px 16px;
  }
  &__icon {
    color: @purple;
    vertical-align: -2px;
  }
  &__title {
    color: #99aab5;
    font-size: 18px;
    font-weight: 500;
  }
  &__divider {
    width: 100%;
    height: 1px;
    margin: 12px 0;
    background-color: #e9e7e7;
  }
  &__more > a {
    color: #99aab5;
    font-weight: 500;
  }
}

.LatestComment {
  &__author {
    font-weight: bold;
  }
  &__bullet:after {
    content: '\2022';
    font-size: 14px;
    margin-left: 4px;
  }
  &__date {
    margin-left: 4px;
    color: #99aab5;
    font-size: 14px;
  }
}
