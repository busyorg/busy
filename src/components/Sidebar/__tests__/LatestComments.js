import React from 'react';
import { shallow } from 'enzyme';
import LatestComments from '../LatestComments';

describe('<LatestComments />', () => {
  it('renders without exploding', () => {
    const props = {
      comments: [
        { id: 0, text: 'hello world', author: 'sekhmet', created: '2017-10-17T09:52:51' },
        { id: 1, text: 'hi Busy', author: 'fabien', created: '2017-10-11T10:52:51' },
      ],
    };
    const wrapper = shallow(<LatestComments {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
