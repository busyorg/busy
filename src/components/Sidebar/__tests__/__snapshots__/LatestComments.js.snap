// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LatestComments /> renders without exploding 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="LatestComments"
  >
    <div
      className="LatestComments__container"
    >
      <h4
        className="LatestComments__title"
      >
        <i
          className="iconfont icon-time LatestComments__icon"
        />
         
        <FormattedMessage
          defaultMessage="Latest Comments"
          id="latest_comments"
          values={Object {}}
        />
      </h4>
      <div
        className="LatestComments__divider"
      />
      <RenderComment
        author="sekhmet"
        created="2017-10-17T09:52:51"
        id={0}
        text="hello world"
      />
      <RenderComment
        author="fabien"
        created="2017-10-11T10:52:51"
        id={1}
        text="hi Busy"
      />
      <div
        className="LatestComments__divider"
      />
      <h4
        className="LatestComments__more"
      >
        <Link
          replace={false}
          to="/latest-comments"
        >
          <FormattedMessage
            defaultMessage="See All Comments"
            id="see_all_comments"
            values={Object {}}
          />
        </Link>
      </h4>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="LatestComments"
    >
      <div
        className="LatestComments__container"
      >
        <h4
          className="LatestComments__title"
        >
          <i
            className="iconfont icon-time LatestComments__icon"
          />
           
          <FormattedMessage
            defaultMessage="Latest Comments"
            id="latest_comments"
            values={Object {}}
          />
        </h4>
        <div
          className="LatestComments__divider"
        />
        <RenderComment
          author="sekhmet"
          created="2017-10-17T09:52:51"
          id={0}
          text="hello world"
        />
        <RenderComment
          author="fabien"
          created="2017-10-11T10:52:51"
          id={1}
          text="hi Busy"
        />
        <div
          className="LatestComments__divider"
        />
        <h4
          className="LatestComments__more"
        >
          <Link
            replace={false}
            to="/latest-comments"
          >
            <FormattedMessage
              defaultMessage="See All Comments"
              id="see_all_comments"
              values={Object {}}
            />
          </Link>
        </h4>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LatestComments
        comments={
          Array [
            Object {
              "author": "sekhmet",
              "created": "2017-10-17T09:52:51",
              "id": 0,
              "text": "hello world",
            },
            Object {
              "author": "fabien",
              "created": "2017-10-11T10:52:51",
              "id": 1,
              "text": "hi Busy",
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "comments": Array [
            Object {
              "author": "sekhmet",
              "created": "2017-10-17T09:52:51",
              "id": 0,
              "text": "hello world",
            },
            Object {
              "author": "fabien",
              "created": "2017-10-11T10:52:51",
              "id": 1,
              "text": "hi Busy",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="LatestComments"
        >
          <div
            className="LatestComments__container"
          >
            <h4
              className="LatestComments__title"
            >
              <i
                className="iconfont icon-time LatestComments__icon"
              />
               
              <FormattedMessage
                defaultMessage="Latest Comments"
                id="latest_comments"
                values={Object {}}
              />
            </h4>
            <div
              className="LatestComments__divider"
            />
            <RenderComment
              author="sekhmet"
              created="2017-10-17T09:52:51"
              id={0}
              text="hello world"
            />
            <RenderComment
              author="fabien"
              created="2017-10-11T10:52:51"
              id={1}
              text="hi Busy"
            />
            <div
              className="LatestComments__divider"
            />
            <h4
              className="LatestComments__more"
            >
              <Link
                replace={false}
                to="/latest-comments"
              >
                <FormattedMessage
                  defaultMessage="See All Comments"
                  id="see_all_comments"
                  values={Object {}}
                />
              </Link>
            </h4>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="LatestComments"
        >
          <div
            className="LatestComments__container"
          >
            <h4
              className="LatestComments__title"
            >
              <i
                className="iconfont icon-time LatestComments__icon"
              />
               
              <FormattedMessage
                defaultMessage="Latest Comments"
                id="latest_comments"
                values={Object {}}
              />
            </h4>
            <div
              className="LatestComments__divider"
            />
            <RenderComment
              author="sekhmet"
              created="2017-10-17T09:52:51"
              id={0}
              text="hello world"
            />
            <RenderComment
              author="fabien"
              created="2017-10-11T10:52:51"
              id={1}
              text="hi Busy"
            />
            <div
              className="LatestComments__divider"
            />
            <h4
              className="LatestComments__more"
            >
              <Link
                replace={false}
                to="/latest-comments"
              >
                <FormattedMessage
                  defaultMessage="See All Comments"
                  id="see_all_comments"
                  values={Object {}}
                />
              </Link>
            </h4>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LatestComments
    comments={
      Array [
        Object {
          "author": "sekhmet",
          "created": "2017-10-17T09:52:51",
          "id": 0,
          "text": "hello world",
        },
        Object {
          "author": "fabien",
          "created": "2017-10-11T10:52:51",
          "id": 1,
          "text": "hi Busy",
        },
      ]
    }
  />,
}
`;
